### PR TITLE
I18n support

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
cf #3 
support of different keyboard layout.

### CODE REVIEW

**Comment by AdrieanKhisbe** [comment]
Just fixed some of the bug I encontered, when there is no `)` in the keyboard layout

**Comment by anmonteiro** [comment]
Thanks, I'll take a look soon

**Comment by fazzone** [comment]
If you want to define the keyboard layout using a string of just the symbols, you can do something like this:
```
(defun create-layout (symbol-row)
  (cl-mapcar (lambda (num sym) (cons (format "%d" num) (make-string 1 sym)))
	    '(1 2 3 4 5 6 7 8 9 0)
	     symbol-row))

(create-layout "!@#$%^&*()")
=> (("1" . "!") ("2" . "@") ("3" . "#") ("4" . "$") ("5" . "%") ("6" . "^") ("7" . "&") ("8" . "*") ("9" . "(") ("0" . ")"))
```
You could potentially have a function that asks people to roll their finger down their number row while holding shift to configure their keyboard layout.  Or you have the string of your keyboard symbols defined in your `.emacs` or something.  Not sure whether that or the current approach makes it easier for people to get snoopy-mode working with whatever keyboard they might be using.

**Comment by AdrieanKhisbe** [comment]
@fazzone I had in mind something like this as an improvement.
Thanks for the snippet of code.
Maybe we should opt in for an hybrid approach with already defined digit-layout, and utilities to define your own just rolling one's fingers. (by define I mean eitheir in the .emacs.d or interactively)

**Comment by AdrieanKhisbe** [comment]
(rebased on top of #6 )

**Comment by AdrieanKhisbe** [comment]
Hey @anmonteiro ! :)
Did you have time to have a look? 
=)

**Comment by AdrieanKhisbe** [comment]
Just added interactive customization method for better experience

**Comment by anmonteiro** [comment]
I'll get to this soon, sorry for the delay

**Comment by AdrieanKhisbe** [comment]
@anmonteiro no problem 😃

File: snoopy.el [hunk 1]
```diff
@@ -66,39 +66,107 @@
...
+
+(defvar snoopy-keyboard-digit-layout-list
+  '(snoopy-qwerty-keyboard-digit-layout
+    snoopy-azerty-fr-osx-keyboard-digit-layout))
+
+(defvar snoopy-qwerty-keyboard-digit-layout "!@#$%^&*()"
+  "Keyboard mapping for qwerty")
+
+(defvar snoopy-azerty-fr-osx-keyboard-digit-layout "&é\"'(§è!çà"
+  "Keyboard mapping for qwerty")
```
Comment (anmonteiro) [inline snoopy.el]: azerty

File: snoopy.el [hunk 2]
```diff
@@ -66,39 +66,107 @@
...
+(defcustom snoopy-lighter " Snoopy"
+  "Mode line lighter for Snoopy Mode."
+  :group 'snoopy
+  :type 'string)
+
+(defvar snoopy-keyboard-digit-layout-list
+  '(snoopy-qwerty-keyboard-digit-layout
+    snoopy-azerty-fr-osx-keyboard-digit-layout))
+
+(defvar snoopy-qwerty-keyboard-digit-layout "!@#$%^&*()"
```
Comment (anmonteiro) [inline snoopy.el]: let's call this `snoopy-qwerty-en-us-keyboard-digit-layout`

File: snoopy.el [hunk 3]
```diff
@@ -66,39 +66,107 @@
...
+  "Keyboard mapping for qwerty")
+
+(defvar snoopy-azerty-fr-osx-keyboard-digit-layout "&é\"'(§è!çà"
+  "Keyboard mapping for qwerty")
+
+(defcustom snoopy-current-layout
```
Comment (anmonteiro) [inline snoopy.el]: let's call this one `snoopy-keyboard-layout`

**Comment by anmonteiro** [comment]
@AdrieanKhisbe sorry for the delay. This looks good to me, just left a couple comments.

This is good to merge after you fix those, unless @fazzone has anything else to say

**Comment by AdrieanKhisbe** [comment]
@anmonteiro No problem :)
I just applied your recommendations. I think this is snoopy 0.2 ready

**Comment by anmonteiro** [comment]
Awesome, thanks!

**Comment by AdrieanKhisbe** [comment]
Super!! 😃

**Pull request merged**